tests/test_state.py

```diff
@@ -2845,3 +2845,41 @@ def bar(self) -> str:
     assert RxState._potentially_dirty_substates() == {State}
     assert State._potentially_dirty_substates() == {C1}
     assert C1._potentially_dirty_substates() == set()
+
+
+@pytest.mark.asyncio
+async def test_setvar(mock_app: rx.App, token: str):
+    """Test that setvar works correctly.
+
+    Args:
+        mock_app: An app that will be returned by `get_app()`
+        token: A token.
+    """
+    state = await mock_app.state_manager.get_state(_substate_key(token, TestState))
+
+    # Set Var in same state (with Var type casting)
+    for event in rx.event.fix_events(
+        [TestState.setvar("num1", 42), TestState.setvar("num2", "4.2")], token
```

Comment:
what is the way to set the num2 to a string value of `"4.2"` here?

Reply:
num2 is typed as a float, you can't (shouldn't) set it to a str

Reply:
oh yeah, should've looked at the hidden part of the code. That's what the type casting is. Sounds good.